lightning/src/chain/utxointerface.rs

```diff
@@ -0,0 +1,24 @@
+//! Trait which allow others parts of rust-lightning to manage CPFP candidates
+//! utxos for increasing feerate of time-sensitive transactions.
+
+
+use bitcoin::blockdata::transaction::OutPoint as BitcoinOutPoint;
+use bitcoin::blockdata::transaction::Transaction;
+
+use chain::package::BumpingOutput;
+
+/// A trait which sould be implemented to provide fresh CPFP utxo for onchain
+/// transactions.
+///
+/// Implementation MUST provision and bookmarked utxo correctly to ensure LN
+/// channel security in case of adversarial counterparty or unfavorable mempool
+/// congestion.
+//TODO: document better
+pub trait UtxoPool: Sync + Send {
+	/// Allocate a utxo to cover fee required to confirm a pending onchain transaction.
+	fn allocate_utxo(&self, required_fee: u64) -> Option<(BitcoinOutPoint, BumpingOutput)>;
```

Comment:
Does `required_fee` also need to cover the witness data for the UTXO? Or is witness data not included in the fee calculation?